runtime/runtime-core/jvm/src/aws/smithy/kotlin/runtime/content/ByteStreamJVM.kt

```diff
@@ -24,11 +24,12 @@ public fun ByteStream.Companion.fromFile(file: File): ByteStream = file.asByteSt
  */
 public fun File.asByteStream(start: Long = 0, endInclusive: Long = length() - 1): ByteStream {
     require(start >= 0) { "start index $start cannot be negative" }
-    require(endInclusive >= start) {
-        "end index $endInclusive must be greater than or equal to start index $start"
-    }
-
     val len = length()
+    if (len != 0L) {
+        require(endInclusive >= start) {
+            "end index $endInclusive must be greater than or equal to start index $start"
+        }
+    }
```

Comment:
**Question**: Couldn't we just go with `require(endExclusive >= start - 1L)` (and no `if` condition)? Applies several other places in this PR.